sdk/storage/storage-file-datalake/src/utils/utils.common.ts

```diff
@@ -595,7 +597,15 @@ export function windowsFileTimeTicksToTime(timeNumber: string | undefined): Date
   // since 12:00 A.M. January 1, 1601 Coordinated Universal Time (UTC).
   // Date accepts a value that represents miliseconds from 12:00 A.M. January 1, 1970
   // Here should correct the year number after converting.
-  const date = new Date(timeNumberInternal / 10000);
+  const timeNumerInMs = timeNumberInternal / 10000;
+  const date = new Date(timeNumerInMs);
+
+  // When initializing date from a miliseconds number the day after 2023-03-01 is still 2023-03-01.
+  // For example, 13322188799999 is 2023-03-01T23:59:59.999Z, while 13322188800000 is 2023-03-01T00:00:00.000Z
+  // Here is to work around the bug.
+  if (timeNumerInMs >= BugTimeBeginningInMS) {
+    date.setUTCDate(date.getUTCDate() + 1);
```

Comment:
Please raise a bug (or check if any exist bug) to NodeJS team, and need remove/refine this code after this bug is fixed.

Reply:
We already had a better fix for this calculation in main prior to this PR: https://github.com/Azure/azure-sdk-for-js/pull/25072